### DISCUSSION TITLE
Dynamically load locale for dayjs

### DISCUSSION THREAD

**Discussion by bencehusi** [mantinedev/mantine#4206] [Q&A] [answered]
I've checked the docs:
https://mantine.dev/dates/dates-provider/

I'm using Next.js (13.3. with app router).

I want to be able to import the locale based on the user's locale `'dayjs/locale/nl';`

All the solutions I came up with failed. For example:

```javascript
'use client';
import { DatesProvider as MantineDatesProvider } from '@mantine/dates';
export function DatesProvider({ children }: { children: React.ReactNode }) {
  const { locale = 'nl' } = useParams();
  import(`dayjs/locale/${locale}.js`)
  return (
    <MantineDatesProvider settings={{ locale, firstDayOfWeek: 1 }}>
      {children}
    </MantineDatesProvider>
  );
}
```

Doing this does NOTHING. The locale is still the default English.

Please, provide guidance on how to do it. All the solutions I came up with failed.

**Comment by rtivital** [comment, marked as answer]
I advise you to import all locales that you will need at the root of your application. dayjs locales file are minimal and do not have any noticeable impact on bundle size. Example file – https://github.com/iamkun/dayjs/blob/dev/src/locale/en.js
